packages/amplication-data-service-generator-runner/src/utils/DynamicPackageInstallationManager.ts

```diff
@@ -1,27 +1,42 @@
 import { valid } from "semver";
+import type { Promisable } from "type-fest";
 import { Tarball } from "./Tarball";
 export class DynamicPackageInstallationManager {
   constructor(private pluginInstallationPath: string) {}
 
-  async installMany(packages: PackageInstallation[]): Promise<void> {
-    for (const plugin of packages) {
-      await this.install(plugin);
+  async install(
+    plugin: PackageInstallation,
+    hooks: {
+      before?: HookFunction;
+      after?: HookFunction;
+      crash?: HookFunction;
```

Comment:
Is this naming conventional? If not, I think we should use something more descriptive like "onBeforeInstall", "onAfterInstall", "onError"